### PR TITLE
Check if we already define ssize_t

### PR DESCRIPTION
**Pull request by talregev**
When compile curl with libuv (for their tests) in vcpkg, I get this error:
```
C:\vcpkg\installed\x64-windows\include\uv\win.h(27,18): error C2628: 'intptr_t' followed by '__int64' is illegal (did you forget a ';'?) [D:\a\curl\curl\bld\src\curl.vcxproj]
Error: Process completed with exit code 1.
```

curl project already define the `ssize_t`.
So I added a `#if defined` that check that.

### CODE REVIEW

**Review by vtjnash — CHANGES_REQUESTED** [review]
ssize_t should be a typedef, so this change should be invalid. A header that wants to define it themselves should be defining `_SSIZE_T_` so that other headers (like this one) can detect it

**Comment by talregev** [comment]
I am not maintainer of vcpkg. I am a user that contribute to vcpkg.
I can make a PR in vcpkg that it will patch it inside vcpkg. 
And I want to let you know about this problem.

**Comment by dg0yt** [comment]
> vcpkg already define the `ssize_t`.

No.

**Comment by talregev** [comment]
> > vcpkg already define the `ssize_t`.
> 
> No.

So I guess it come from curl project.

**Comment by dg0yt** [comment]
Not directly, but maybe from one of its dependencies. 
`grep -R "def.* ssize_t" installed/x64-linux/include/` 
has a lot hits, and IIUC there are POSIX headers which may provide the type.

**Comment by talregev** [comment]
> Not directly, but maybe from one of its dependencies. `grep -R "def.* ssize_t" installed/x64-linux/include/` has a lot hits, and IIUC there are POSIX headers which may provide the type.

I can think of other implementation for upstream:
```c
# ifdef ssize_t
# define _SSIZE_T_
# define _SSIZE_T_DEFINED
# endif
```

**Comment by dg0yt** [comment]
If `ssize_t` is expected to come with `_SSIZE_T_DEFINED` or `_SSIZE_T` on windows (IDK), you should identify and fix the header which leaks an inconsistent pattern. AFAICS the inconsistent definition doesn't come from the headers included by libuv, so you must check consuming sources.

**Comment by talregev** [comment]
Thank you for your advice. I will not have time to do such investigation.

**Comment by bnoordhuis** [comment]
I'm going to go ahead and close this because it can't land as-is. Thanks anyway.

**Comment by talregev** [comment]
@dg0yt Did you explore that problem?

**Comment by vtjnash** [comment]
You might also consider adding it to the CFLAGS or defines of your project, so that you don't need to alter libuv to workaround the other header causing issues

**Comment by talregev** [comment]
I can patch it in vcpkg.

**Comment by dg0yt** [comment]
> @dg0yt Did you explore that problem?

The next curl release will use `uv.h` not only in examples but also `src/tool_operate.c`. You will probably have to explore a few more depepencies for the source of the problem.

**Comment by dg0yt** [comment]
Sortcut: 

> So I guess it come from curl project.

It can come from curl indeed, subject to `CMakeLists.txt`:
https://github.com/curl/curl/blob/552d32886b193d5b6846be42fd2362298779f484/lib/curl_config.h.cmake#L789-L790
This file isn't installed, only used while building curl. It probably didn't hurt much as long as `uv.h` was used by an example only, but this changes now.

**Pull request closed without merging**
